components/Square.tsx

```diff
@@ -15,47 +19,104 @@ const colors = {
     green: "#00FF00"
 }
 
+const notifyMessage = (msg: string) => {
+    if (Platform.OS === 'android') {
+        ToastAndroid.showWithGravity(msg, ToastAndroid.SHORT, ToastAndroid.CENTER)
+        console.log("Toast shown.");
+    } else {
+        Alert.alert(msg);
+    }
+}
+
 const jewelStyle = (option: string): { backgroundColor: string } => {
     switch (option){
-        case 'A':
+        case '1':
             return {
                 backgroundColor: colors.red
             }
-        case 'B':
+        case '2':
             return {
                 backgroundColor: colors.blue
             }
-        case 'C':
+        case '3':
             return {
                 backgroundColor: colors.yellow
             }
-        case 'D':
+        case '4':
             return {
                 backgroundColor: colors.green
             }
         default:
             return {
                 backgroundColor: colors.grey
             }
-            
     }
 }
 
 export default class Square {
     type: string;
     key: number;
-    constructor(props: { title: string; key: number}){
+    swap: (location: number, direction: string) => Square[];
+    constructor(props: { title: string; key: number, swap: (location: number, direction: string) => Square[]}){
         this.type = props.title; 
         this.key = props.key;
+        this.swap = props.swap;
+     }
+
+     onSwipe(direction: String, gestureState: object) {
+        console.log("You swiped square " + this.type);
+        console.log("  at position " + this.key);
+        console.log("  direction is "+ direction);
+        console.log();
      }
 
-     render () {
+     setKey(newKey: number) {
+         this.key = newKey;
+     }
+
+     render () {    
+        const config = {
+            velocityThreshold: 0.3,
```

Comment:
store this config in a separate config file so it can be imported in and settings are easily changed.